reflex/app.py

```diff
@@ -633,10 +649,14 @@ def add_custom_404_page(
             on_load: The event handler(s) that will be called each time the page load.
             meta: The metadata of the page.
         """
-        if component is None:
-            component = Default404Page.create()
+        console.deprecate(
+            feature_name="App.add_custom_404_page",
+            reason="Use App.add_page(component, route=rx.constants.Page404.SLUG) instead.",
```

Comment:
```suggestion
            reason=f"Use app.add_page(component, route={rx.constants.Page404.SLUG!r}) instead.",
```

i dont think we should have the deprecation message show this kind of weird undocumented constant. the value is always going to be `/404`, so lets just show that